lib/src/calculators/calculate_crop_fit_params.dart

```diff
@@ -0,0 +1,126 @@
+import 'dart:math';
+
+import 'package:custom_image_crop/src/models/params_model.dart';
+import 'package:custom_image_crop/src/widgets/custom_image_crop_widget.dart';
+
+/// Returns params to use for displaying crop screen.
+CropFitParams calculateCropFitParams({
+  required double screenWidth,
+  required double screenHeight,
+  required double cropPercentage,
+  required int imageWidth,
+  required int imageHeight,
+  required CustomImageFit imageFit,
+  required double aspectRatio,
+}) {
+  /// the width of the area to crop
+  final double cropSizeWidth;
+
+  /// the height of the area to crop
+  final double cropSizeHeight;
+
+  /// used to adjust image scale
+  final double defaultScale;
+
+  switch (imageFit) {
+    case CustomImageFit.fillCropSpace:
```

Comment:
I think I would also move this to separate classes

Reply:
I disagree, it's much easier to see the other implementations directly 🤔